pkg/controller/mpi_job_controller.go

```diff
@@ -639,7 +639,7 @@ func (c *MPIJobController) syncHandler(key string) error {
 			}
 		}
 		if launcher == nil {
-			if mpiJob.Spec.LauncherCreationPolicy == kubeflow.LauncherCreationPolicyAtStartup || c.countReadyWorkerPods(worker) == len(worker) {
+			if mpiJob.Spec.LauncherCreationPolicy == kubeflow.LauncherCreationPolicyAtStartup || (!isMPIJobSuspended(mpiJob) && c.countReadyWorkerPods(worker) == len(worker)) {
```

Comment:
This looks wrong too. If the job is suspended, we shouldn't create the launcher, even if the policy is `AtStartup`.

cc @tenzen-y 

I think this issue has been there for a while?

Reply:
Oh, I see, we still create the job, but suspend it https://github.com/kubeflow/mpi-operator/blob/edbde8caeb09c4d778761c8a1b1307474f832bef/pkg/controller/mpi_job_controller.go#L1445

Reply:
> Oh, I see, we still create the job, but suspend it
> 
> https://github.com/kubeflow/mpi-operator/blob/edbde8caeb09c4d778761c8a1b1307474f832bef/pkg/controller/mpi_job_controller.go#L1445

Yes, so I think such an issue doesn't exist.